src/Kestrel.Core/Internal/Http2/Http2FrameWriter.cs

```diff
@@ -32,6 +32,20 @@ public Http2FrameWriter(PipeWriter outputPipeWriter, PipeReader outputPipeReader
             _outputReader = outputPipeReader;
         }
 
+        public void Complete()
+        {
+            lock (_writeLock)
+            {
+                if (_completed)
+                {
+                    return;
+                }
+
+                _completed = true;
+                _outputWriter.Complete();
```

Comment:
This logic will need further improvement, but the needed changes are pretty large so I filed #2672. As part of fixing this, I imagine that the Complete and Abort methods here will look more like [those in Http1OutputProducer](https://github.com/aspnet/KestrelHttpServer/blob/14f020cf11c0d79472c9080064c8899e52c24b3c/src/Kestrel.Core/Internal/Http/Http1OutputProducer.cs#L146-L192).